adapters/IronSource/IronSourceAdapter/GADMAdapterIronSourceRtbInterstitialAd.m

```diff
@@ -0,0 +1,193 @@
+     onLog:[NSString stringWithFormat:@"Showing IronSource interstitial ad for Instance ID: %@",
+            self.instanceID]];
+    id<GADMediationInterstitialAdEventDelegate> interstitialDelegate = self.interstitialAdEventDelegate;
+
+    if (!self.biddingISAInterstitialAd){
+        if (interstitialDelegate){
+            NSError *error = GADMAdapterIronSourceErrorWithCodeAndDescription(
+                                                                              GADMAdapterIronSourceErrorFailedToShow,
+                                                                              @"the ad is nil");
+            [interstitialDelegate didFailToPresentWithError:error];
+        }
+        [GADMAdapterIronSourceUtils
+         onLog:[NSString stringWithFormat:@"Failed to show due to ad not loaded, for Instance ID: %@",
+                self.instanceID]];
+        return;
+    }
+    
+    [self.biddingISAInterstitialAd setDelegate:self];
+    [self.biddingISAInterstitialAd showFromViewController:viewController ];
+}
+
+
+- (void)interstitialAd:(nonnull ISAInterstitialAd *)interstitialAd didFailToShowWithError:(nonnull NSError *)error {
+    [GADMAdapterIronSourceUtils
+     onLog:[NSString stringWithFormat:@"%@ with error= %@", NSStringFromSelector(_cmd), error.localizedDescription]];
+    id<GADMediationInterstitialAdEventDelegate> interstitialDelegate = self.interstitialAdEventDelegate;
+    if (!interstitialDelegate){
+        return;
+    }
+    [interstitialDelegate didFailToPresentWithError:error];
+}
+
+- (void)interstitialAdDidLoad:(nonnull ISAInterstitialAd *)interstitialAd {
+    [GADMAdapterIronSourceUtils
+     onLog:[NSString stringWithFormat:@"%@ instanceId= %@ adId= %@", NSStringFromSelector(_cmd),
+            interstitialAd.adInfo.instanceId,
+            interstitialAd.adInfo.adId]];
+    
+    self.biddingISAInterstitialAd = interstitialAd;
+    if (!self.interstitalAdLoadCompletionHandler){
+        return;
+    }
+    
+    self.interstitialAdEventDelegate = self.interstitalAdLoadCompletionHandler(self, nil);
+}
+
+- (void)interstitialAdDidFailToLoadWithError:(nonnull NSError *)error {
+    [GADMAdapterIronSourceUtils
+     onLog:[NSString stringWithFormat:@"%@ with error= %@ ", NSStringFromSelector(_cmd),
+            error.localizedDescription]];
+    if (!self.interstitalAdLoadCompletionHandler){
+        return;
+    }
+    self.interstitalAdLoadCompletionHandler(nil, error);
+    
+}
+
+- (void)interstitialAdDidShow:(nonnull ISAInterstitialAd *)interstitialAd {
+    [GADMAdapterIronSourceUtils
+     onLog:[NSString stringWithFormat:@"%@ instanceId= %@ adId= %@", NSStringFromSelector(_cmd),
+            interstitialAd.adInfo.instanceId,
+            interstitialAd.adInfo.adId]];
+    id<GADMediationInterstitialAdEventDelegate> interstitialDelegate = self.interstitialAdEventDelegate;
+    if (!interstitialDelegate){
+        return;
+    }
+    [interstitialDelegate willPresentFullScreenView];
```

Comment:
In the cases where the SDK doesn't have a separate callback for this event, we invoke this callback at the end of presentFromViewController. So, I suggest moving this there.

Reply:
Got it, updated